server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java

```diff
@@ -154,13 +156,26 @@ public static void resolveAndValidateRouting(Metadata metadata, String concreteI
     @Override
     protected void doExecute(Task task, final UpdateRequest request, final ActionListener<UpdateResponse> listener) {
         if (request.isRequireAlias() && (clusterService.state().getMetadata().hasAlias(request.index()) == false)) {
-            throw new IndexNotFoundException(
+            IndexNotFoundException e = new IndexNotFoundException(
                 "[" + DocWriteRequest.REQUIRE_ALIAS + "] request flag is [true] and [" + request.index() + "] is not an alias",
                 request.index()
             );
+
+            incDocStatusStats(e);
+            throw e;
+        }
+
+        boolean shouldAutoCreate;
+
+        try {
+            shouldAutoCreate = autoCreateIndex.shouldAutoCreate(request.index(), clusterService.state());
+        } catch (Exception e) {
+            incDocStatusStats(e);
+            throw e;
```

Comment:
This is not really document level error right? This is more like index metadata management which failed. It shouldnt be counted in doc level status